packages/api/lib/writeRecords/write-execution.js

```diff
@@ -117,15 +119,17 @@ const _writeExecutionRecord = ({
   esClient,
 }) => knex.transaction(async (trx) => {
   logger.info(`About to write execution ${postgresRecord.arn} to PostgreSQL`);
-  const [executionCumulusId] = await executionPgModel.upsert(trx, postgresRecord);
-  logger.info(`Successfully wrote execution ${postgresRecord.arn} to PostgreSQL with cumulus_id ${executionCumulusId}`);
+  const [response] = await executionPgModel.upsert(trx, postgresRecord);
+  const translatedExecution = await translatePostgresExecutionToApiExecution(response, trx);
+  logger.info(`Successfully wrote execution ${postgresRecord.arn} to PostgreSQL with cumulus_id ${response.cumulus_id}`);
   await writeExecutionToDynamoAndES({
     dynamoRecord,
     executionModel,
     updatedAt,
     esClient,
   });
-  return executionCumulusId;
+  await publishExecutionSnsMessage(translatedExecution);
```

Comment:
I'm trying to decide if we want to do this inside the transaction, because it means that if the publishing to SNS were to throw an error for some reason, then the PG write would be rolled back (but Dynamo and ES) are not. 

I'm leaning towards not doing it in the transaction and instead calling this from `writeExecutionRecordFromMessage` after `_writeExecutionRecord`. 

Either way we go, `publishExecutionSnsMessage` doesn't get reached unless all the writes succeed, but the difference is that not having it in the transaction won't complicate the rollback of the transaction any further. But it might be good to get other team opinions on this question

Reply:
I think that's a compelling reason to move it outside of the transaction

Reply:
[`bd0112b` (#2422)](https://github.com/nasa/cumulus/pull/2422/commits/bd0112b50cc0fda564ec0be0a84a830659daf1ea)

Reply:
Confirmed with the team that we don't want to publish to SNS in the transaction